stan/math/rev/mat/fun/gp_periodic_cov.hpp

```diff
@@ -230,18 +225,13 @@ class gp_periodic_cov_vari<T_x, double, T_l, T_p> : public vari {
         sigma_d_(sigma),
         p_d_(value_of(p)),
         sigma_sq_d_(sigma_d_ * sigma_d_),
-        dist_(ChainableStack::instance_->memalloc_.alloc_array<double>(
-            size_ltri_)),
-        sin_2_dist_(ChainableStack::instance_->memalloc_.alloc_array<double>(
-            size_ltri_)),
-        sin_dist_sq_(ChainableStack::instance_->memalloc_.alloc_array<double>(
-            size_ltri_)),
+        dist_(stack_mem.alloc_array<double>(size_ltri_)),
+        sin_2_dist_(stack_mem.alloc_array<double>(size_ltri_)),
+        sin_dist_sq_(stack_mem.alloc_array<double>(size_ltri_)),
         l_vari_(l.vi_),
         p_vari_(p.vi_),
-        cov_lower_(ChainableStack::instance_->memalloc_.alloc_array<vari *>(
-            size_ltri_)),
-        cov_diag_(
-            ChainableStack::instance_->memalloc_.alloc_array<vari *>(size_)) {
+        cov_lower_(stack_mem.alloc_array<vari *>(size_ltri_)),
+        cov_diag_(stack_mem.alloc_array<vari *>(size_)) {
```

Comment:
[optional]

I'd prefer consistency of `vari*` rather than having `vari *` in some places.